test/src/helpers.cc

```diff
@@ -857,6 +858,110 @@ void write_array(
   tiledb_config_free(&cfg);
 }
 
+void write_array_without_key(
```

Comment:
Same here, use only one function `write_array` and remove any occurrence of the deprecated APIs from everywhere in the code.